preflight-packages/examples.jetstack.io/gke_basic/gke.rego

```diff
@@ -4,42 +4,50 @@ package gke_basic
 import input.gke.Cluster as gke
 
 # Rule 'private_cluster'
-default preflight_private_cluster = false
-preflight_private_cluster {
-        gke.privateClusterConfig.enablePrivateNodes == true
+preflight_private_cluster[message] {
+	not gke.privateClusterConfig.enablePrivateNodes
+
+	message := "cluster does not have private nodes enabled"
 }
 
 # Rule 'basic_auth_disabled'
-default preflight_basic_auth_disabled = false
-preflight_basic_auth_disabled {
-        not gke.masterAuth.username
-        not gke.masterAuth.password
+preflight_basic_auth_disabled[message] {
+	# masterAuth must be missing or an empty {}
+	{ gke.masterAuth } & { null, {}} == set()
+
+	message := "cluster does not have basic auth disabled"
 }
 
 # Rule 'abac_disabled'
-default preflight_abac_disabled = false
-preflight_abac_disabled {
-        not gke.legacyAbac.enabled == true
+preflight_abac_disabled[message] {
+	gke.legacyAbac.enabled
+
+	message := "cluster has legacy abac enabled"
 }
 
 # Rule 'k8s_master_up_to_date'
-default preflight_k8s_master_up_to_date = false
-preflight_k8s_master_up_to_date {
-        re_match(`^1\.1[234].*$`, gke.currentMasterVersion)
+preflight_k8s_master_up_to_date[message] {
+	not gke.currentMasterVersion
```

Comment:
I think `currentMasterVersion` will always have a value. This was using regex to check it started `1.12`, `1.13` or `.14`. Probably needs updating but something we should check.

Reply:
So this is ok to leave as is? The old tests seem to account for this so I left it in.

Reply:
I don't think this would ever not have a value, the old test may have just been unnecessary. https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#Cluster.FIELDS.current_master_version

Reply:
Sounds good, I've removed in e21905e0929073018328ea4b91ea106d0d4ad16a